master/src/main/java/org/apache/celeborn/service/deploy/master/clustermeta/ha/HARaftServer.java

```diff
@@ -498,19 +490,19 @@ public void updateServerRole() {
       RaftProtos.RaftPeerRole thisNodeRole = roleInfoProto.getRole();
 
       if (thisNodeRole.equals(RaftProtos.RaftPeerRole.LEADER)) {
-        setServerRole(thisNodeRole, masterRatisAddress.getAddress().toString());
-
+        setServerRole(thisNodeRole, getRpcEndpoint());
       } else if (thisNodeRole.equals(RaftProtos.RaftPeerRole.FOLLOWER)) {
         ByteString leaderNodeId = roleInfoProto.getFollowerInfo().getLeaderInfo().getId().getId();
         // There may be a chance, here we get leaderNodeId as null. For
         // example, in 3 node Ratis, if 2 nodes are down, there will
         // be no leader.
-        String leaderPeerAddress = null;
+        String leaderPeerRpcEndpoint = null;
         if (leaderNodeId != null && !leaderNodeId.isEmpty()) {
-          leaderPeerAddress = roleInfoProto.getFollowerInfo().getLeaderInfo().getId().getAddress();
+          leaderPeerRpcEndpoint =
+              roleInfoProto.getFollowerInfo().getLeaderInfo().getId().getClientAddress();
         }
 
-        setServerRole(thisNodeRole, leaderPeerAddress);
+        setServerRole(thisNodeRole, leaderPeerRpcEndpoint);
```

Comment:
I made a mistake in https://github.com/alibaba/celeborn/pull/752.

Before #752, only the host is reserved, and since there is a contract that all master nodes use the same port, so it's sufficient. but the tricky things are, the ratis and rpcEnv could bind different ip/host, the assumption is fragile then.

after #752, the function is broken, since ratis port is combined w/ the host, in this PR, I fixed it to the rpcPort and change the name to "*RpcEndpoint" to make it more clear.